src/robocoup/commands/expertise.js

```diff
@@ -324,9 +254,156 @@ addCommand('find', {
+    const dialog = new Dialog({
+      channel,
+      timeout: 60,
+      onTimeout: `Timed out, please type \`${command}\` to try again.`,
+      onCancel: () => {
+        return `Canceled, please type \`${command}\` to try again.`;
+      },
+    });
+
+    function ask(_oneTimeHeader) {
+      const newValues = {};
+      return dialog.questions({
+        oneTimeHeader: _oneTimeHeader,
+        question: ({exit, timeout}) => `Please choose your interest level for ${expertiseName}:`,
+        choices: INTEREST,
+        onMatch: match => {
+          newValues.interest = match;
+          return `_You selected *${newValues.interest}* for interest, thanks!_`;
+        },
+      }, {
+        question: ({exit, timeout}) => `Please choose your experience level for ${expertiseName}:`,
+        choices: EXPERIENCE,
+        onMatch: match => {
+          newValues.experience = match;
+          return `_You selected *${newValues.experience}* for experience, thanks!_`;
+        },
+      }, () => oldValues && {
+        question: ({exit, timeout}) => `Why has your experience/interest changed for ${expertiseName}?`,
+        onResponse: reason => {
+          newValues.reason = reason;
+          return '_Noted!_';
+        },
+      }, {
+        question: ({exit, timeout}) => {
+          const reason = 'reason' in newValues ? `> Reason: *${newValues.reason}*\n` : '';
+          return heredoc.trim.unindent`
+            You've entered the following for ${expertiseName}. Is this ok?
+
+            > Interest: *${INTEREST[newValues.interest - 1]}* (${newValues.interest})
+            > Experience: *${EXPERIENCE[newValues.experience - 1]}* (${newValues.experience})
+            ${reason}
+          `;
+        },
+        choices: [
+          `Save these changes.`,
+          `No, re-choose interest and experience for ${expertiseName}.`,
+        ],
+        onMatch(match) {
+          if (match === 2) {
+            return ask();
+          }
+          return updateExpertise({user, expertise, newValues}).then(done);
+        },
+      });
+    }
+    let lastUpdated;
+    if (oldValues) {
+      const formatted = moment.duration(-oldValues.seconds_since_last_update, 'seconds').humanize(true);
+      lastUpdated = `_You last updated this expertise *${formatted}*._`;
+    }
+    return ask([
+      oneTimeHeader,
+      lastUpdated,
+      '',
+      `> ${expertise.description}`,
+    ]);
+  });
+}
+
+function updateMissing({channel, user}) {
+  function ask(header) {
+    return query('expertise_missing_by_bocouper', user)
+    .then(missing => {
+      if (missing.length === 0) {
+        return heredoc.trim.unindent`
+          You have no outstanding expertise data.
+          View your expertise list with \`expertise me\`.
+        `;
+      }
+      const expertise = missing[0];
+      const num = missing.length === 1 ? '' : ` ${missing.length}`;
+      const oneTimeHeader = [
+        ...(header ? [header, ''] : []),
```

Comment:
Yeah, this code is insane. I think I'm going to need to make a full commenting pass after this is all done, but I'm not sure I want to hold up merging and deploying this code to do so. If I feel pressed for time, I'll merge this as-is and create an issue to comment more thoroughly.